maven-enforcer-plugin/src/main/java/org/apache/maven/plugins/enforcer/EnforceMojo.java

```diff
@@ -153,6 +155,25 @@ public class EnforceMojo extends AbstractMojo {
     @Component
     private EnforcerRuleManager enforcerRuleManager;
 
+    /**
+     * List of strings that matches the EnforcerRules to execute.
+     *
+     * @deprecated Use <code>enforcer.rules</code> property instead
+     */
+    @Parameter(required = false, property = "rules")
+    @Deprecated
+    public void setCommandLineRules(List<String> rulesToExecute) {
+        if (rulesToExecute != null && !rulesToExecute.isEmpty()) {
+            this.rulesToExecute = rulesToExecute;
+        }
+    }
+
+    public void setRulesToExecute(List<String> rulesToExecute) {
+        if (rulesToExecute != null && !rulesToExecute.isEmpty()) {
+            this.rulesToExecute = rulesToExecute;
+        }
+    }
```

Comment:
The only disadvantage of this approach is that we don't really have a direct control of which method is called first, so if both properties are specified the behavior could be potentially different (from what I was able to observe, the `setCommandLineRules` is getting called first, but I am not sure if that's guaranteed every time).

Reply:
Yes we don't have control which method will be called first.

but we can throw exception if `rulesToExecute` is already set with not empty list,
with information that only one parameter should be used - what do you think?

We need both setters form technical reason to have two parameters, but we can simply call `setRulesToExecute` from `setCommandLineRules` to have logic in one place - next my idea 😄